### ISSUE TITLE
feature request: option to not die when unrecognized flags are passed

### ISSUE THREAD

**Issue by bricerebsamen** [gflags/gflags#148]
That would help making gflags work with some other option parsing libraries (boost::program_options for instance).

Since gflags can remove the flags it recognizes, then whatever is left on the command line can be parsed manually and the user can decide whether it belongs to another library or it's an error.

```
int main(int argc, char **argv) {
  gflags::ParseCommandLineFlags(&argc, &argv, true, /*allow_unregistered*/ true);
  some_other_parser.parse(&argc, &argv);
  //etc.
  // whatever remains on the command line, is supposed to be positional argument
  // it one of them starts with - or -- then it's an error
}
```

**Comment by tomasz-wiszkowski** [comment]
Have you tried passing `--undefok=any_flag,that_may_not_exist,and_more` ?

this should do the following:
- if the flag is known (defined): process the flag
- if the flag is unknown (undefined): ignore the flag and carry on.

**Comment by schuhschuh** [comment]
Any such feature that does not require to pass additional options should also consider #61.

**Comment by srivastavag** [comment]
What is the current status on this ?

**Comment by bricerebsamen** [comment]
I went the other way round: I wrote a library that removes the boost options first, and then pass the remaining arguments to gflags. That works for now. Also I am migrating away from boost ;)

**Comment by srivastavag** [comment]
@bricerebsamen Yes I am also thinking of this, however I am not convinced that dying if there are unknown flags is the correct behaviour, at the least there should be some programmatic way (ie other than undefok on command line, IMO somewhat awkward for end user to expose how a flag library should work) of telling gflags to either ignore unknown flags or to allow program to continue after giving warning.

**Comment by schuhschuh** [comment]
Have you tried calling [AllowCommandLineReparsing()](https://github.com/gflags/gflags/blob/aa2d0f7b4ed11849f40c3a1dc573ac331ebdaebd/src/gflags.h.in#L369) before `ParseCommandLineFlags`?

This sets the internal boolean switch [allow_command_line_reparsing](https://github.com/gflags/gflags/blob/aa2d0f7b4ed11849f40c3a1dc573ac331ebdaebd/src/gflags.cc#L1272) that silences errors.

**Comment by srivastavag** [comment]
@schuhschuh Thanks a ton! I will explore this, hopefully it would work. :-)
